internal/server/tls.go

```diff
@@ -25,21 +22,47 @@ func tlsConfigFromOptions(
 	tlsCacheDir string,
 	opts TLSOptions,
 ) (*tls.Config, error) {
-	// TODO: print CA fingerprint when the client can trust that fingerprint
+	// TODO: how can we test this?
+	if opts.ACME {
+		if err := os.MkdirAll(tlsCacheDir, 0o700); err != nil {
+			return nil, fmt.Errorf("create tls cache: %w", err)
+		}
 
-	if opts.Certificate != "" && opts.PrivateKey != "" {
-		roots, err := x509.SystemCertPool()
-		if err != nil {
-			logging.Warnf("failed to load TLS roots from system: %v", err)
-			roots = x509.NewCertPool()
+		manager := &autocert.Manager{
+			Prompt: autocert.AcceptTOS,
+			Cache:  autocert.DirCache(tlsCacheDir),
+			// TODO: according to the docs HostPolicy should be set to prevent
```

Comment:
We should open an issue for this, dont want it to get overlooked

Reply:
https://github.com/infrahq/infra/issues/2484, I'll add it to the comment as well.